### PR TITLE
Clarify `ShopifyApp::Authenticated` usage in docs

### PR DESCRIPTION
**Pull request by nelsonwittwer**
### What this PR does

Adds clarifying docs on how to avoid common problems with `ShopifyApp::Authenticated` and App Bridge 3

### Things to focus on

Confirm docs are clear and accurate

### Checklist

Before submitting the PR, please consider if any of the following are needed:

- [x] Update `CHANGELOG.md` if the changes would impact users
- [x] Update `README.md`, if appropriate.
- [x] Update any relevant pages in `/docs`, if necessary
- [x] For security fixes, the [Disclosure Policy](https://github.com/Shopify/shopify_app/blob/master/SECURITY.md#disclosure-policy) must be followed.

### CODE REVIEW

File: docs/Troubleshooting.md [hunk 1]
```diff
@@ -144,13 +143,9 @@ X-Shopify-API-Request-Failure-Unauthorized: true
 
 Then, use the [Shopify App Bridge Redirect](https://shopify.dev/tools/app-bridge/actions/navigation/redirect) action to redirect your app frontend to the app login URL if this header is set.
 
-## Migrating to App Bridge 2.0
 
-In order to upgrade your embedded app to the latest App Bridge 2.0 version, please refer to the [migration guide](https://shopify.dev/tutorials/migrate-your-app-to-app-bridge-2).
+### I'm stuck in a redirect loop after OAuth
 
-To ensure that your app's embedded layout doesn't import App Bridge 2.0 before fully migrating, make the following change to bind it to v1.x.
+In previous versions of `ShopifyApp::Authenticated` controller concern and App Bridge embededd apps were able to include the `Authenticated` controller concern in the `HomeController` and other embeded controllers. This is no longer supported due to security concerns with third party cookies. App Bridge 3 is needed to handle all embeded sessions.
```
Comment (mkevinosullivan) [inline docs/Troubleshooting.md:149]:
```suggestion
In previous versions of `ShopifyApp::Authenticated` controller concern, App Bridge embedded apps were able to include the `Authenticated` controller concern in the `HomeController` and other embedded controllers. This is no longer supported due to browsers blocking 3rd party cookies to increase privacy. App Bridge 3 is needed to handle all embedded sessions.
```
(includes Paulo's suggested change)

File: docs/shopify_app/authentication.md [hunk 2]
```diff
@@ -106,6 +106,10 @@ end
 
 For backwards compatibility, the engine still provides a controller called `ShopifyApp::AuthenticatedController` which includes the `ShopifyApp::Authenticated` concern. Note that it inherits directly from `ActionController::Base`, so you will not be able to share functionality between it and your application's `ApplicationController`.
 
+#### Embededd apps and `ShopifyApp::Authenticated`
+
+Embedded Shopify Admin apps are unable to use the `ShopifyApp::Authenticated` controller concern. Those who inclue this concern in the `HomeController` or some other embdeded controller will see what looks like an OAuth redirect loop as the `ShopifyApp::Authenticated` concern will be fighting with the App Bridge. For more details on how to handle embeded sessions, refer to [the session token documentation](https://shopify.dev/apps/auth/oauth/session-tokens).
```
Comment (mkevinosullivan) [inline docs/shopify_app/authentication.md:111]:
```suggestion
#### Embedded apps and `ShopifyApp::Authenticated`

Embedded Shopify Admin apps can only use the `ShopifyApp::Authenticated` controller concern *if* the requests originate from App Bridge's `authenticatedFetch` method. Those who include this concern in the `HomeController` or some other embedded controller will see what looks like an OAuth redirect loop as the `ShopifyApp::Authenticated` concern will be fighting with the App Bridge. For more details on how to handle embedded sessions, refer to [the session token documentation](https://shopify.dev/apps/auth/oauth/session-tokens).
```

**Comment by andyw8** [comment]
Related: https://github.com/Shopify/shopify_app/issues/1502

**Comment by nelsonwittwer** [comment]
@paulomarg @mkevinosullivan - Thanks for the great suggestions! Any other concerns y'all have with merging this in?

**Review by mkevinosullivan — APPROVED** [review]

**Pull request merged**